### PR TITLE
PR for Swift/T Release 1.3

### PR DESCRIPTION
**Pull request by jmjwozniak**

### CODE REVIEW

File: var/spack/repos/builtin/packages/adlbx/package.py [hunk 1]
```diff
@@ -30,11 +30,11 @@ class Adlbx(AutotoolsPackage):
...
 
     homepage = 'http://swift-lang.org/Swift-T'
-    url      = 'http://swift-lang.github.io/swift-t-downloads/adlbx-0.8.0.tar.gz'
-    version('0.8.0', '34ade59ce3be5bc296955231d47a27dd')
+    url      = 'http://swift-lang.github.io/swift-t-downloads/1.3/spack/adlbx-0.8.1.tar.gz'
+    version('0.8.1', 'c63a08315170e80d57816d7af92d27db')
 
     depends_on('exmcutils')
-    depends_on('mpi')
+    depends_on('mpich')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/adlbx/package.py]: It doesn't work with any other MPI provider?

File: var/spack/repos/builtin/packages/turbine/package.py [hunk 2]
```diff
@@ -30,17 +30,30 @@ class Turbine(AutotoolsPackage):
...
                 '--with-tcl='     + self.spec['tcl'].prefix,
                 '--with-mpi='     + self.spec['mpi'].prefix]
+        if '+python' in self.spec:
+            if self.spec['python'].satisfies('@3'):
+                # newer python3.X does not create python link by default
+                pythonex = 'python3'
+            else:
+                pythonex = 'python'
+            args += ['--with-python-exe='
+                     + join_path(self.spec['python'].prefix.bin, pythonex)]
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/turbine/package.py]: You can replace this entire thing with:
```python
if '+python' in self.spec:
    args.append('--with-python-exe={0}'.format(self.spec['python'].command.path))
```

File: var/spack/repos/builtin/packages/adlbx/package.py (hunk 1 above)
Comment (jmjwozniak) [inline var/spack/repos/builtin/packages/adlbx/package.py]: That is correct.

**Comment by jmjwozniak** [comment]
The old URLs are still in place.  We moved the distribution package TGZs to a separate GitHub repo to limit the growth in the size of our source code repo.

File: var/spack/repos/builtin/packages/turbine/package.py (hunk 2 above)
Comment (bd4) [inline var/spack/repos/builtin/packages/turbine/package.py]: Thanks that is much cleaner, we will make the change. I had trouble finding things like this in the package documentation, which mentions only prefix.bin and version: https://spack.readthedocs.io/en/latest/packaging_guide.html#accessing-dependencies .

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/turbine/package.py]: Yes, documentation is a bit of a work in progress. I expect to see this feature documented in #3377. There will probably be a rush to get all of our documentation up-to-date right before SC17.

File: var/spack/repos/builtin/packages/adlbx/package.py (hunk 1 above)
Comment (tgamblin) [inline var/spack/repos/builtin/packages/adlbx/package.py]: @jmjwozniak: Just for my personal edification, what are the special MPICH features it uses?

Comment (jmjwozniak) [inline var/spack/repos/builtin/packages/adlbx/package.py]: Swift/T is a MPI-based workflow system that is (as of 1.3) capable of calling MPI programs. MPICH allows itself to be invoked recursively, where another MPI implementation we tried does not.

File: var/spack/repos/builtin/packages/turbine/package.py (hunk 2 above)
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/turbine/package.py]: Ping @jmjwozniak. Once this code block is replaced, Travis should pass.

File: var/spack/repos/builtin/packages/turbine/package.py [hunk 3]
```diff
@@ -30,17 +30,25 @@ class Turbine(AutotoolsPackage):
...
+    depends_on('python', when='+python')
 
     def configure_args(self):
         args = ['--with-c-utils=' + self.spec['exmcutils'].prefix,
                 '--with-adlb='    + self.spec['adlbx'].prefix,
                 '--with-tcl='     + self.spec['tcl'].prefix,
                 '--with-mpi='     + self.spec['mpi'].prefix]
+        if '+python' in self.spec:
+            args.append('--with-python-exe={0}'.format(
+                                            self.spec['python'].command.path))
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/turbine/package.py]: There is a Flake8 error on this line:
```
var/spack/repos/builtin/packages/turbine/package.py:53: [E126] continuation line over-indented for hanging indent
```
I believe you want this line to be indented 4-spaces more than the previous line. You can confirm by running `spack flake8`.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/turbine/package.py]: Ping @jmjwozniak

**Comment by alalazo** [comment]
@jmjwozniak Can you rebase this branch, and ping me back?

**Comment by jmjwozniak** [comment]
Hi- it looks like codecov/project is blocking this PR: is this a real error?  If not, please review and merge this PR for the new Swift/T 1.4.3 .  Thanks

**Comment by adamjstewart** [comment]
@jmjwozniak See the note at the bottom of https://spack.readthedocs.io/en/latest/contribution_guide.html#coverage

**Comment by jmjwozniak** [comment]
Ok great- am I queued for review?

**Comment by scheibelp** [comment]
Thanks for the PR and the reviews!

File: var/spack/repos/builtin/packages/stc/package.py [hunk 4]
```diff
@@ -30,14 +30,14 @@ class Stc(AutotoolsPackage):
...
+    url      = 'http://swift-lang.github.io/swift-t-downloads/spack/stc-0.0.0.tar.gz'
 
-    version('0.7.3', '6bf769f406f6c33d1c134521373718d3')
+    version('0.8.2', '883b0657f1aac9b81158ef0a8989be4c')
 
     depends_on('java')
     depends_on('ant')
     depends_on('turbine')
-    depends_on('zsh', type='run')
+    depends_on('zsh')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/stc/package.py]: I'm guessing that many of these are either `'build'` or `('build', 'run')` dependencies and not `('build', 'link')` as omitting the `type` implies.  Is that correct?

File: var/spack/repos/builtin/packages/turbine/package.py [hunk 5]
```diff
@@ -30,18 +30,41 @@ class Turbine(AutotoolsPackage):
...
 
+    variant('python', default=False,
+            description='Enable calling python')
+    variant('r', default=False,
+            description='Enable calling R')
+    depends_on('adlbx@:0.8.0', when='@:1.1.0')
+    depends_on('adlbx', when='@1.2.1:')
     depends_on('adlbx')
     depends_on('tcl')
     depends_on('zsh')
```
Comment (tgamblin) [inline var/spack/repos/builtin/packages/turbine/package.py]: same question here -- does this link with some lib from zsh or tcl?

**Comment by jmjwozniak** [comment]
Please review again after 5afbc0c .  turbine does link against tcl but that is handled by our build system so I did not specify type='link' .

**Comment by jmjwozniak** [comment]
Since this was merged 3 days ago, do I need to create a new PR?

**Comment by scheibelp** [comment]
Yes

**Comment by jmjwozniak** [comment]
The new PR is at: https://github.com/spack/spack/pull/9537

**Pull request merged**